.github/workflows/ci.yml

```diff
@@ -139,3 +133,25 @@ jobs:
       rootless-docker: false
       run-tests: true
       ryuk-disabled: false
+
+  sonarcloud:
+    if: ${{ github.ref_name == 'main' && github.repository_owner == 'testcontainers' }}
+    needs: test-examples
```

Comment:
Do we need to wait until all modules&examples are run? I think we could `need` the main job.

In follow-up PRs, where we add support for the modules, we could move this out to the end (or have one sonarcloud job for each job type: core, modules, examples...) 🤔

Reply:
There is no obligation but I thought that having only one step to upload every report was better than having one per module.

Reply:
Yes, indeed. I meant one job per type: all the modules, all the examples, not for each module or example. I would not consider examples for code coverage, as they are just simple code snippets, not real modules exposing API

Reply:
Hum... that will make the workflow quite more complex as the upload-artifact will need to receive a dedicated name depending on module, modulegen or root .

Reply:
Ok, one last question: what if the build failed in any of the modules? this step won't be executed. i think we should make it mandatory (as a post-action) for the entire workflow